packages/framework/fluid-static/src/treeRootDataObject.ts

```diff
@@ -0,0 +1,216 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	BaseContainerRuntimeFactory,
+	TreeDataObject,
+	TreeDataObjectFactory,
+} from "@fluidframework/aqueduct/internal";
+import type { IDataObjectProps } from "@fluidframework/aqueduct/internal";
+import type {
+	IContainerRuntimeOptions,
+	MinimumVersionForCollab,
+} from "@fluidframework/container-runtime/internal";
+import type {
+	IContainerRuntime,
+	IContainerRuntimeInternal,
+} from "@fluidframework/container-runtime-definitions/internal";
+import type {
+	FluidObject,
+	IFluidHandle,
+	IFluidLoadable,
+} from "@fluidframework/core-interfaces";
+import { assert, fail } from "@fluidframework/core-utils/internal";
+import type { IChannelFactory } from "@fluidframework/datastore-definitions/internal";
+import type { IFluidDataStoreRegistry } from "@fluidframework/runtime-definitions/internal";
+import type { SharedObjectKind } from "@fluidframework/shared-object-base/internal";
+import { SharedTreeFactoryType } from "@fluidframework/tree/internal";
+
+import { compatibilityModeRuntimeOptions } from "./compatibilityConfiguration.js";
+import { createDataObject, createSharedObject } from "./rootDataObject.js";
+import type {
+	CompatibilityMode,
+	ContainerSchema,
+	IRootDataObject,
+	IStaticEntryPoint,
+	LoadableObjectKind,
+	LoadableObjectRecord,
+} from "./types.js";
+import {
+	compatibilityModeToMinVersionForCollab,
+	isDataObjectKind,
+	isSharedObjectKind,
+	makeFluidObject,
+} from "./utils.js";
+
+interface IProvideTreeRootDataObject {
+	readonly TreeRootDataObject: TreeRootDataObject;
+}
+
+interface TreeRootDataObjectExtraProps {
+	readonly treeKey: string;
+}
+
+/**
```

Comment:
TODO: update docs in this module to note tree-based distinction from `rootDataObject`.